app/controllers/recipes_controller.rb

```diff
@@ -2,10 +2,15 @@ class RecipesController < ApplicationController
 
   def new
     @recipe = Recipe.new
+    3.times { @recipe.grains.build }
   end
 
   def create
     @recipe = current_user.recipes.create(recipe_params)
+    @recipe.grains.each do |grain|
+      grain.weight.to_f
```

Comment:
I removed this since the recipes are saving without it (by some sort of rails magic). Unless you think I should include it anyway...